### PR TITLE
Custom response objects

### PR DESCRIPTION
**Pull request by novitae**
This is a PR related to #397

Here is the test I did:
```py
>>> from curl_cffi.requests import AsyncSession, Session, Response
>>> import asyncio
>>> 
>>> class MyResponse(Response):
...     def __init__(self, *args, **kwargs):
...         super().__init__(*args, **kwargs)
...     @property
...     def status(self):
...         return self.status_code
... 
>>> async def main():
...     response = await AsyncSession(response_factory=MyResponse).get("https://www.lego.com")
...     print("AsyncSession with MyResponse as custom factory ->", type(response), response.status)
...     response = await AsyncSession().get("https://www.lego.com")
...     print("AsyncSession without custom factory ->", type(response))
...     response = Session(response_factory=MyResponse).get("https://www.lego.com")
...     print("Session with MyResponse as custom factory ->", type(response), response.status)
...     response = Session().get("https://www.lego.com")
...     print("Session without custom factory ->", type(response))
... 
>>> if __name__ == "__main__":
...     asyncio.run(main())
... 
AsyncSession with MyResponse as custom factory -> <class '__main__.MyResponse'> 200
AsyncSession without custom factory -> <class 'curl_cffi.requests.models.Response'>
Session with MyResponse as custom factory -> <class '__main__.MyResponse'> 200
Session without custom factory -> <class 'curl_cffi.requests.models.Response'>
```

### CODE REVIEW

**Comment by lexiforest** [comment]
I like the idea! Could you please add a few unittests in `/tests` and examples in `/examples`?

**Comment by novitae** [comment]
> I like the idea! Could you please add a few unittests in `/tests` and examples in `/examples`?

Done !

File: curl_cffi/requests/session.py [hunk 1]
```diff
@@ -245,6 +248,13 @@ def __init__(
         self.interface = interface
         self.cert = cert
 
+        if response_class is None:
+            response_class = Response
+        elif issubclass(response_class, Response) is False:
```
Comment (lexiforest) [inline curl_cffi/requests/session.py:253]:
```diff
- elif issubclass(response_class, Response) is False:
+ elif not issubclass(response_class, Response):
```

File: tests/integration/test_response_class.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    session = requests.Session(response_class=CustomResponse)
+    response = session.get("http://example.com")
+    assert isinstance(response, CustomResponse)
+    assert hasattr(response, "status")
+    print(response.status)
+
+class WrongTypeResponse: pass
+
+def test_wrong_type_custom_response():
+    try:
```
Comment (lexiforest) [inline tests/integration/test_response_class.py:23]: It's better to use:

```py
with pytest.raises(TypeError):
    ...
```

See: https://docs.pytest.org/en/7.1.x/how-to/assert.html#assertions-about-expected-exceptions

File: examples/custom_response_class.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from curl_cffi import requests
+
+class CustomResponse(requests.Response):
+    @property
```
Comment (lexiforest) [inline examples/custom_response_class.py:4]: Coud you please add an `__init__` method that extracts other values from the `.curl` property.

Something like this:

https://github.com/lexiforest/curl_cffi/blob/137a83451e8032d6ffaed1f54fa605f828760309/curl_cffi/requests/session.py#L706-L707

For a list of getinfo options, see here: https://curl.se/libcurl/c/easy_getinfo_options.html

File: curl_cffi/requests/session.py (hunk 1 above)
Comment (novitae) [reply curl_cffi/requests/session.py]: Done

File: tests/integration/test_response_class.py (hunk 2 above)
Comment (novitae) [reply tests/integration/test_response_class.py]: Done

File: examples/custom_response_class.py (hunk 3 above)
Comment (novitae) [reply examples/custom_response_class.py:11]: Done. By the way, any `CurlInfo` of `0x600000` (such as `SIZE_UPLOAD_T`) aren't supported in this dict:
https://github.com/lexiforest/curl_cffi/blob/137a83451e8032d6ffaed1f54fa605f828760309/curl_cffi/curl.py#L232-L237
And cause a `KeyError` when reaching:
https://github.com/lexiforest/curl_cffi/blob/137a83451e8032d6ffaed1f54fa605f828760309/curl_cffi/curl.py#L243

**Comment by lexiforest** [comment]
LGTM

**Comment by novitae** [comment]
Merge ?

**Pull request merged**